src/NServiceBus.Core/Routing/UnicastPublishRouter.cs

```diff
@@ -48,7 +50,11 @@ HashSet<string> SelectDestinationsForEachEndpoint(IDistributionPolicy distributi
                 }
                 else
                 {
-                    var subscriber = distributionPolicy.GetDistributionStrategy(group.First().Endpoint, DistributionStrategyScope.Publish).SelectReceiver(group.Select(s => s.TransportAddress).ToArray());
+                    var endpointName = @group.First().Endpoint;
+
+                    var instances = endpointInstances.FindInstances(endpointName).Select(e => transportAddressTranslation(e)).ToArray();
```

Comment:
so instead of using the provided transport address of the subscription, we're just trying to make that up based on the endpoint name?
If you do not configure sender-side distribution, we have no chance of identifying the correct machine or addresses which don't match the endpoint name?